### DISCUSSION TITLE
is it able to create a common/public funtion for all cmds ?

### DISCUSSION THREAD

**Discussion by superiums** [gokcehan/lf#1529] [Q&A] [answered]
some function is needed by lots of cmds, eg: check destionation dir/file exist or not , and interactive with user.
for satisfy rule DONT REPEAT YOURSELF ,  i wonder if  there's a way to define a common/pulic function for all cmds .

**Comment by CatsDeservePets** [comment, marked as answer]
You actually can! There is an example in the [tips](https://github.com/gokcehan/lf/wiki/Tips#use-environment-variables-in-command-or-mapping) section of the Wiki.

**Comment by superiums** [comment]
thanks for your suggestion.
this is a solution as we may write a function string to an env var, then eval it if needed. but not so perfect in bash. but it's perfect for [lumesh](https://github.com/superiums/lumesh) of my own solution. as it could store fn directly as an env var.
Thanks !

**Reply by CatsDeservePets** [reply]
No problem at all!
